node/signer_test.go

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package node
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/config/node"
+)
+
+func TestDefaultConfigInitializationUsesExistingDefaultKey(t *testing.T) {
+	tempDir := t.TempDir()
+	t.Setenv("HOME", tempDir)
+
+	require := require.New(t)
+
+	config := node.DefaultSignerConfig{
+		SignerKeyPath: filepath.Join(tempDir, ".avalanchego/staking/signer.key"),
+	}
+
+	signer1, err := newStakingSigner(config)
+	require.NoError(err)
+	signer2, err := newStakingSigner(config)
+	require.NoError(err)
```

Comment:
Since `newStakingSigner` is a non-exported helper, ideally we wouldn’t test it directly since it's an implementation detail of `node`, not a part of its api. Would it be possible to cover this logic by going through `node.New` instead? That would make the test more realistic and aligned with how the code is actually used by the caller.

Reply:
Testing the idempotence of creating a "default signer" by creating an entire `node` seems completely antithetical to to idea of unit testing. I can push down the [localsigner creation logic](https://github.com/ava-labs/avalanchego/blob/signers-config-wip/node/signer.go#L64-L68) into an exported helper in `localsigner` if that makes the separation cleaner